lib/galaxy/tool_util/deps/__init__.py

```diff
@@ -133,8 +140,8 @@ def __init__(self, default_base_path, conf_file=None, app_config=None):
         else:
             plugin_source = self.__build_dependency_resolvers_plugin_source(conf_file)
         self.dependency_resolvers = self.__parse_resolver_conf_plugins(plugin_source)
-        self._enabled_container_types = []
-        self._destination_for_container_type = {}
+        self._enabled_container_types: List[str] = []
+        self._destination_for_container_type: Dict[str, str] = {}
```

Comment:
```suggestion
        self._destination_for_container_type: Dict[str, Dict[str, "JobDestination"]] = {}
```

with `JobDestination` to be imported `from galaxy.jobs` in a `if TYPE_CHECKING:` .